src/senselab/audio/tasks/speaker_embeddings/api.py

```diff
@@ -0,0 +1,44 @@
+"""This module implements some utilities to extract speaker embeddings from a model."""
+
+from typing import Optional
+
+import pydra
+from speechbrain.inference.speaker import EncoderClassifier
+from torch import Tensor
+
+from senselab.audio.data_structures.audio import Audio
+from senselab.utils.data_structures.device import DeviceType
+from senselab.utils.data_structures.model import HFModel, SenselabModel
+
+
+def extract_embeddings(audio: Audio, model: SenselabModel, device: Optional[DeviceType] = None) -> Tensor:
```

Comment:
- can you please make this `extract_speaker_embeddings_from_audios`?
- can you please make this work with a list of audios?

Reply:
- are CPU, CUDA, and MPS supported? if not, please use the same logic as in the task `speech_to_text` to handle this

Reply:
- i know that for now all models come from speechbrain. however, in the future we may want to integrate models from other libraries (e.g., `espnet`, `pyannote.audio`). hence, can you please create a python file specific for speech brain and make the API method here a wrapper of that method? same as in the `speech_to_text` task. thank you!

Reply:
- last, can you add in the documentation of the speechbrain function that the default threshold for determining if two audios are from the same or different speakers is 0.25 [ref: https://speechbrain.readthedocs.io/en/latest/API/speechbrain.inference.speaker.html#speechbrain.inference.speaker.SpeakerRecognition.verify_batch]

Reply:
Where should I add this? Would it not be better to put that in the speaker verification documentation?

Reply:
Currently, we don't have speaker verification as a task. Speaker verification is generally performed by computing cosine similarity between speaker embeddings. I would add the reference here for now but agree with you that we may want to create a workflow/task for speaker verification.